packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -247,6 +254,29 @@ abstract class HttpApiBase extends Resource implements IHttpApi { // note that t
 
     return vpcLink;
   }
+
+  /**
+   * @internal
+   */
+  public _addIntegration(config: HttpRouteIntegrationConfig): HttpIntegration {
+    const configKey = JSON.stringify(config);
+    if (configKey in this.httpIntegrations) {
+      return this.httpIntegrations[configKey];
+    }
+
+    const integration = new HttpIntegration(this, `${this.node.id}-Integration`, {
```

Comment:
I think the construct id is not accurate. This would cause failure with multiple integrations. Updating it similar to vpcLink